frontend/setupChelonia.js

```diff
@@ -310,6 +310,7 @@ export default ((() => {
   return () => {
     if (!promise) {
       promise = setupChelonia().catch((e) => {
+        console.error('[setupChelonia] Error during chelonia setup', e)
```

Comment:
> I don't see where setupChelonia() is called from the browser.

That's because `setupChelonia()` is only called from the SW. The last PR was about getting Chelonia to run in the SW.


> Therefore I also don't see where a page refresh is attempted or any sort of recovery or even alert()

`service-worker.js` now sends a `ready` message to the SW. The `ready` message calls `setupChelonia()` (in the SW). If setting up the SW failed, `main.js` will attempt to refresh the page. But also, refreshing the page will send another `ready` message, meaning that `setupChelonia()` will be called again. If it failed last time, it will attempt to reconfigure Chelonia.

Reply:
OK, then I don't see how this PR really accomplishes much, as the `'ready'` message that is sent each time a page loads won't do anything since `setupChelonia()` is called in `sw-primary.js` and results in a singleton... calling it again won't do anything...

Reply:
This PR does a few things:

1. It adds logging
2. It ensures that `setupChelonia()` has been called and waits for it.
3. `setupChelonia()` is a singleton only if it succeeded or is in progress. Calling it multiple times does work if it failed.